src/spiral/experimental.clj

```diff
@@ -31,6 +31,11 @@
             (async/>! (:async-response req) shed-response)))))
     req-chan))
 
+(defn- strip-guard-prefix [path guard]
```

Comment:
Could you add a docstring that this returns `nil` if `path` is `nil` or if the `path` would be `nil` after stripping the guard prefix. I feel like that comment would help explain why, on line 50, the `or` is needed for `:uri` but not `:path-info`. Thanks! Other than that, this is good to merge.